arches_for_science/pkg/extensions/functions/multicard_resource_descriptor.py

```diff
@@ -0,0 +1,42 @@
+from arches.app.functions.primary_descriptors import AbstractPrimaryDescriptorsFunction
+from arches.app.models.system_settings import settings
+
+from django.utils.translation import get_language, gettext as _
+
+details = {
+    "functionid": "00b2d15a-fda0-4578-b79a-784e4138664b",
+    "name": "Multi-card Resource Descriptor",
+    "type": "primarydescriptors",
+    "description": "Configure resource descriptors from multiple cards",
+    "defaultconfig": {
+        "descriptor_types": {
+            "name": {"string_template": ""},
+            "map_popup": {"string_template": ""},
+            "description": {"string_template": ""},
+        }
+    },
+    "classname": "MulticardResourceDescriptor",
+    "component": "views/components/functions/multicard-resource-descriptor",
+}
+
+
+class MulticardResourceDescriptor(AbstractPrimaryDescriptorsFunction):
+    """Implemented in the database via triggers on tiles table.
+    
+    This implementation just fetches the calculated result from the db."""
+
+    def get_primary_descriptor_from_nodes(self, resource, config, context=None, descriptor="name"):
+        # TODO: patch core arches to send the appropriate descriptor argument, e.g. "name" or "description" etc.
```

Comment:
@chiatt We should consider patching this in Arches 7.5. As it stands, this function has no knowledge of what descriptor was passed in (which we need if we want to be able to read the JSON field from the db and look up the given descriptor by key).